packages/desktop/src/app.js

```diff
@@ -299,13 +311,26 @@ const Channel = () => {
   const serverMembersByUserId = state.selectServerMembersByUserId(
     params.serverId
   );
-  const channelMessages = state.selectChannelMessages(params.channelId);
+  const messages = state.selectChannelMessages(params.channelId);
+
+  const sortedMessages = messages.sort(
+    (m1, m2) => new Date(m1.created_at) - new Date(m2.created_at)
+  );
+
+  const mostRecentMessage = sortedMessages.slice(-1)[0];
 
   // Fetch messages when switching channels
   React.useEffect(() => {
     actions.fetchMessages({ channelId: params.channelId });
   }, [actions.fetchMessages, params.channelId]);
 
+  // Mark channel as read whenever the user enters a channel, or whenever a new
+  // message appears
```

Comment:
Didn’t think this through, just an example.

Reply:
depending on how scrolling works when stuff is unread, this can probably work

Reply:
getting 2 `channel-mark-read` events at the same time when posting a new message

Reply:
duplication seems to happen also when switching channels

Reply:
Yeah that make sense. When posting a message we first put the optimistic client side one, and then the real one arrives fron the server, so the most recent message changes. 

Same on channel switch since we refetch redundantly, the most recent message reference changes, triggering it twice.